src/main/kotlin/com/mineinabyss/looty/features/recipes/RecipeCraftingSystem.kt

```diff
@@ -27,4 +47,34 @@ class RecipeCraftingSystem : Listener {
             inventory.result = null
         }
     }
+
+    @EventHandler
+    fun PrepareSmithingEvent.onCustomSmithingTransform() {
+        // Smithing will cache the last recipe, so even with 0 input
+        // recipe will return as not null if say a Diamond Hoe was put in before
+        if (inventory.contents.any { it?.isEmpty != false }) return
+        // Return if no item is custom, as then vanilla should handle it fine
+        if (inventory.contents.none { it?.fastPDC?.hasComponentsEncoded == true }) return
+
+        val (template, mineral) = (inventory.inputTemplate ?: return) to (inventory.inputMineral ?: return)
+        val equipment = inventory.inputEquipment ?: return
+
+        val inputGearyEntity = equipment.fastPDC?.decodePrefabs()?.firstOrNull() ?: return
+        val smithingTransformRecipes = Bukkit.recipeIterator().asSequence().filter { (it as? SmithingTransformRecipe)?.result?.fastPDC?.hasComponentsEncoded == true }.filterIsInstance<SmithingTransformRecipe>()
+        val customRecipeResult = smithingTransformRecipes.filter { it.template.test(template) && it.addition.test(mineral) && it.base.itemStack.itemMeta?.persistentDataContainer?.decodePrefabs()?.firstOrNull() == inputGearyEntity }.firstOrNull()?.result
+        var recipeResultItem = (customRecipeResult ?: ItemStack.empty()).let { result?.toSerializable()?.toItemStack(it, EnumSet.of(SerializableItemStack.Properties.DISPLAY_NAME)) }
+
+        recipeResultItem = recipeResultItem?.editItemMeta {
```

Comment:
Is this part necessary? the toItemStack should override as needed and keep the displayName if the ignored property is set. Is this maybe covering some case I'm not considering?

Reply:
newResult is applied ontop of oldResult
this is because otherwise the finalResult will have the prefabs from oldResult aka Diamond Sickle
this means however that the finalResult will have the displayname of oldResult aka Diamond Sickle, or if renamed the renamed

so it does indeed require to fix that afterwards from my testing to get correct behaviour